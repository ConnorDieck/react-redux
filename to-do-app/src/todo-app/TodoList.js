import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
	const INITIAL_STATE = [
		// { id: uuid(), width: 10, height: 10, backgroundColor: "red" },
		// { id: uuid(), width: 20, height: 20, backgroundColor: "blue" }
	];

	const [ todos, setTodos ] = useState(INITIAL_STATE);

	const addTodo = newTodo => {
		setTodos(todos => [ ...todos, { ...newTodo, id: uuid() } ]);
	};
	const remove = todoId => {
		setTodos(todos.filter(todo => todo.id !== todoId));
	};

	return (
		<div>
			<NewTodoForm addTodo={addTodo} />
			<div>
				{todos.map(({ id, text }) => (
					<Todo //
						id={id}
						text={text}
						key={id}
						handleRemove={remove}
					/>
				))}
			</div>
		</div>
	);
};

export default TodoList;
