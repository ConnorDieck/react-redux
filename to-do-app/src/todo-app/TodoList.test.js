import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, text = "wash feet") {
	const textInput = todoList.getByLabelText("New Todo");

	fireEvent.change(textInput, { target: { value: text } });

	const button = todoList.getByText("Submit");
	fireEvent.click(button);
}

it("renders without crashing", function() {
	render(<TodoList />);
});

it("matches snapshot", function() {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});

it("shows a todo after submitting", function() {
	const todoList = render(<TodoList />);

	// First, expect nothing
	expect(todoList.queryByText("Remove Todo")).not.toBeInTheDocument();

	addTodo(todoList);

	const removeBtn = todoList.getByText("Remove Todo");

	expect(removeBtn).toBeInTheDocument();
	expect(removeBtn.previousSibling).toContainHTML(`<div class="Todo">wash feet</div>`);

	expect(todoList.getAllByDisplayValue("")).toHaveLength(1);
});

it("can remove a todo", function() {
	const todoList = render(<TodoList />);
	addTodo(todoList);

	const removeBtn = todoList.getByText("Remove Todo");

	fireEvent.click(removeBtn);
	expect(removeBtn).not.toBeInTheDocument();
});
