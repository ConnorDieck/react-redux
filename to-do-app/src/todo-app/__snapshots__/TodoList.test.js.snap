// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="text"
      >
        New Todo
      </label>
      <input
        id="text"
        name="text"
        placeholder="Todo"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
